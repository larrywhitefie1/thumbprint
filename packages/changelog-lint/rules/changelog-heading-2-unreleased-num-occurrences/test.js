const test = require('../../test-utils');
const rule = require('./');

const valid = ['## Unreleased'];

const invalid = [
    '## UNRELEASED',
    `## Unreleased

## Unreleased`,
];

test({
    rule,
    valid,
    invalid,
});
