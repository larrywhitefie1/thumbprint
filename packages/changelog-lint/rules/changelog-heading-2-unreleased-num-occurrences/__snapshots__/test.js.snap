// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "## UNRELEASED",
  "output": Array [
    Object {
      "column": null,
      "fatal": false,
      "line": null,
      "location": Object {
        "end": Object {
          "column": null,
          "line": null,
        },
        "start": Object {
          "column": null,
          "line": null,
        },
      },
      "reason": "Changelog must contain one second level heading that says “Unreleased“ (0 found).",
      "ruleId": "changelog-heading-2-unreleased-num-occurrences",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "## Unreleased

## Unreleased",
  "output": Array [
    Object {
      "column": null,
      "fatal": false,
      "line": null,
      "location": Object {
        "end": Object {
          "column": null,
          "line": null,
        },
        "start": Object {
          "column": null,
          "line": null,
        },
      },
      "reason": "Changelog must contain one second level heading that says “Unreleased“ (2 found).",
      "ruleId": "changelog-heading-2-unreleased-num-occurrences",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
